framework/src/media/audio/audio_manager.c

```diff
@@ -504,7 +444,7 @@ static int get_avail_audio_in_card_id()
 {
 	int i;
 	for (i = 0; i < CONFIG_AUDIO_MAX_INPUT_CARD_NUM; i++) {
-		if (g_audio_in_cards[i].status == AUDIO_CARD_READY) {
+		if (g_audio_in_cards[i].status != AUDIO_CARD_IDLE) {
```

Comment:
We also need to set initially cards as "AUDIO_CARD_IDLE" when found in "found_audio_card()".

Reply:
If we set AUDIO_CARD_IDLE in found_audio_card, currently there's should be no active card.
I think we have to establish policy for this, I set AUDIO_CARD_IDLE in found_audio_card as you suggested, and I will put TODO above of this function & put basic logic.